frontend/src/store/addProfileModal.ts

```diff
@@ -0,0 +1,34 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import { ModalVisibility } from "interfaces/modalVisibility";
+
+const initialState: ModalVisibility = {
+    status: "",
+    visibility: false,
+};
+
+export const changeAddProfileVisibility = createAsyncThunk(
+    "addProfileModalVisibility/changeAddProfileVisibility",
+    (visibility: boolean) => {
+        return visibility ? true : false;
```

Comment:
`return !!visibility`

Reply:
though not sure if you need an async thunk for that, since it's a simple state change - normal reducer should be able to change this